proposals/apple-mdm/base-setup-clix.md

```diff
@@ -0,0 +1,205 @@
+# MDM Base Setup
+
+Note: these can all be namespaced behind something like `apple-mdm` if needed. But I would like to avoid creating an mdm vertical vs. an osquery vertical to create an integrated product.
+
+## APNS
+### Step 1 - Generate CSR
+``` sh
+$ fleetctl setup apns init
+Generating MDM keys at ./
+  fleet-apns-pki.crt ... done
+  fleet-apns-pki.key ... done
+  fleet-apns-push.key ... done
+  fleet-apns-push.csr ... done
+Send the .csr and .crt files to your friendly Fleet representative
+
+Keep the .key files. You will need them to complete this process. 
+
+Fleet will return a signed certificate to you within 1 business day. Use that file with the .key files in the `fleetctl setup apns gen-req` command.
+```
+
+### Step 2 - Generate .req file from .p7
+``` bash
+$ fleetctl setup apns gen-req \
+--p7-file=~/path/to/*.p7 \
+--pki-key=~/path/to/fleet-apns-pki.key \
+--pki-crt=~/path/to/fleet-apns-pki.crt
+Working...
+Generated fleet-apns-push.req file at ./
+
+Visit identity.apple.com to upload this .req file. For more details, follow these instructions: NEED LINK HERE
+
+Apple will generate a .crt file. Use that file with the .key files in the `fleetctl setup apns finalize` command. 
+```
+
+### Step 3 - Upload push..crt and push.key files to Fleet ENV
+
+```sh
+$ fleetctl setup apns finalize \
+--apple-crt=/path/to/apple.crt \
+--push-key=/path/to/fleetctl-apns-push.key \
+--context=dogfood
+This command will update your Fleet Server's APNS environment variables with the content of the specified files. 
+
+Continue? (Y/n)
+
+Updating ENV["NEED_VARNAME"] with contents of /path/to/apple.crt
+Updating ENV["NEED_VARNAME"] with contents of /path/to/fleetctl-apns-push.key
+```
+
+## SCEP
+### Step 1 - Generate SCEP CA keys
+```bash
+$ fleetctl setup scep-ca init
+Generating SCEP CA and keys at ./
+  fleet-scep.crt ... done
+  fleet-scep.key ... done
+  
+Next, use these file in the `fleetctl setup scep-ca finalize` command.
+```
+
+### Step 2 - Update fleet server ENV vars with new certs
+```bash
+$ fleetctl setup scep-ca finalize \ 
+--scep-crt=/path/to/fleet-scep.crt \
+--scep-key=/path/to/fleet-scep.key \
+--context=dogfood
+This command will update your Fleet Server's SCEP environment variables with the content of the specified files. 
+
+WARNING: If you change the SCEP CA in the fleet server, you must un-enroll and re-enroll all devices.
+
+Continue? (Y/n)
+
+Updating ENV["NEED_VARNAME"] with contents of /path/to/fleet-scep.crt
+Updating ENV["NEED_VARNAME"] with contents of /path/to/fleet-scep.key
+```
+
+## Misc
+### Error States
+#### Missing required flag
+Matches current behavior of `fleetctl`: Simply shows the help page (try `fleetctl setup --name="hello"`)
+#### Required file does not exist 
+Matches current behavior of `fleetctl`: `Error: open does/not/exist: no such file or directory`
+(Try `fleetctl apply -f does/not/exist`)
```

Comment:
Is this "(Try `fleetctl apply...`)" suggestion helpful when running these `fleetctl setup` commands? 

If not, I propose removing the "(Try `fleetctl apply -f does/not/exist`)" part of the error message from the current behavior. What do you think?

I think this part of the message makes more sense when running `fleetctl apply -f`. That being said, I'm not sure how helpful this part of the message is here either...

Reply:
Oh shoot. that's poor writing on my part. This is an instruction to the reader of this document, but should not show up in the error message. Basically, I'm telling you, you can get this error message by running `fleetctl apply -f adsfasdf`. Will fix to make clearer.

Reply:
Ah, got it!